packages/optimizely-sdk/lib/core/custom_attribute_condition_evaluator/index.js

```diff
@@ -115,16 +140,28 @@ function existsEvaluator(condition, userAttributes) {
  * Evaluate the given greater than match condition for the given user attributes
  * @param   {Object}    condition
  * @param   {Object}    userAttributes
+ * @param   {Object}    logger
  * @returns {?Boolean}  true if the user attribute value is greater than the condition value,
  *                      false if the user attribute value is less than or equal to the condition value,
  *                      null if the condition value isn't a number or the user attribute value
  *                      isn't a number
  */
-function greaterThanEvaluator(condition, userAttributes) {
-  var userValue = userAttributes[condition.name];
+function greaterThanEvaluator(condition, userAttributes, logger) {
+  var conditionName = condition.name;
+  var userValue = userAttributes[conditionName];
   var conditionValue = condition.value;
 
-  if (!fns.isFinite(userValue) || !fns.isFinite(conditionValue)) {
+  if (!fns.isFinite(conditionValue)) {
+    return null;
+  }
+
+  if (userValue === null) {
+    logger.log(LOG_LEVEL.WARNING, sprintf(LOG_MESSAGES.UNEXPECTED_TYPE_NULL, MODULE_NAME, JSON.stringify(condition), conditionName));
+    return null;
+  }
+
+  if (!fns.isFinite(userValue)) {
+    logger.log(LOG_LEVEL.WARNING, sprintf(LOG_MESSAGES.UNEXPECTED_TYPE, MODULE_NAME, JSON.stringify(condition), conditionName, userValue));
```

Comment:
```suggestion
    logger.log(LOG_LEVEL.WARNING, sprintf(LOG_MESSAGES.UNEXPECTED_TYPE, MODULE_NAME, JSON.stringify(condition), conditionName, userValueType));
```
where `userValueType = typeof userValue`.

(Same thing in `lessThanEvaluator` and `substringEvaluator`.)

Reply:
Nope, this test case has been written for exact condition only, which does not have the ordering issue. I am adding the same test case for other conditions now.